tests/lax_numpy_test.py

```diff
@@ -1674,6 +1674,26 @@ def testPolySub(self, a_shape, b_shape, dtype):
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True)
 
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}_order={}_k={}".format(
+          jtu.format_shape_dtype_string(a_shape, dtype),
+          order, k),
+       "dtype": dtype, "a_shape": a_shape, "order" : order, "k": k}
+      for dtype in default_dtypes
+      for a_shape in one_dim_array_shapes
+      for order in range(5)
+      for k in [np.arange(order), 0., 1., order, None]
+))
+  def testPolyInt(self, a_shape, order, k, dtype):
+    rng = jtu.rand_default(self.rng())
+    np_fun = lambda arg1: np.polyint(arg1, m=order, k=k)
+    jnp_fun = lambda arg1: jnp.polyint(arg1, m=order, k=k)
+    args_maker = lambda: [rng(a_shape, dtype)]
+    tol = {np.float16: 2e-1, np.float32: 1e-2, np.float64: 1e-8}
```

Comment:
I'm surprised that the tolerance needs to be this loose. Do you understand why this is necessary?

Reply:
I got tests failing with precision errors using the recursive approach. However, now everything seems to be fine using the default tolerances.